src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -1161,17 +1161,12 @@ bool ciMethod::was_executed_more_than(int times) {
 // ------------------------------------------------------------------
 // ciMethod::has_unloaded_classes_in_signature
 bool ciMethod::has_unloaded_classes_in_signature() {
-  VM_ENTRY_MARK;
-  {
-    ExceptionMark em(THREAD);
-    methodHandle m(THREAD, get_Method());
-    bool has_unloaded = Method::has_unloaded_classes_in_signature(m, thread);
-    if( HAS_PENDING_EXCEPTION ) {
-      CLEAR_PENDING_EXCEPTION;
-      return true;     // Declare that we may have unloaded classes
-    }
-    return has_unloaded;
-  }
+  // ciSignature is resolved against some accessing class and
+  // signature classes aren't required to be local. As a benefit,
+  // it makes signature classes visible through loader constraints.
+  // So, encountering an unloaded class signals it is absent both in
+  // the callee (local) and caller contexts.
```

Comment:
```suggestion
  // the callee (local) and caller contexts, because linking loads signature classes.
```

Reply:
Can you elaborate, please? Neither method resolution nor method selection steps trigger loading of signature classes.

Reply:
Maybe "loads signature classes" is the wrong choice of words.  I mean what SystemDictionary::check_signature_loaders() does.  Using the CI interface is guaranteed to go through SystemDictionary::check_signature_loaders(), right?

Reply:
CI doesn't use `SystemDictionary::check_signature_loaders()`. `ciSignature` constructor eventually calls into `SystemDictionary::find_constrained_instance_or_array_klass()` which also tries `LoaderConstraintTable::find_constrained_klass()` if local search fails.

I tried to stress that by stating that "signature classes aren't required to be local" and "it makes signature classes visible through loader constraints".

Reply:
OK, thanks for the explanation.